packages/preview/acrostiche-0.2.0/acrostiche.typ

```diff
@@ -0,0 +1,126 @@
+          }else if acronyms.at(acr).len() == 1{
+            acronyms.at(acr).at(0)+"s"
+          }else{
+            acronyms.at(acr).at(1)
+          }
+        }else{
+          panic("Definitions should be arrays of one or two strings. Definition of "+acr+ " is: "+type(defs))
+        }
+      }else{
+        panic(acr+" is not a key in the acronyms dictionary.")
+      }
+    })
+  }else{
+    state("acronyms",none).display(acronyms=>{
+      if acr in acronyms{
+        let defs = acronyms.at(acr)
+        if type(defs) == "string"{ // If user forgot the trailing comma the type is string
+          defs
+        }else if type(defs)== "array"{
+          if acronyms.at(acr).len() == 0{panic("No definitions found for acronym "+acr+". Make sure it is defined in the dictionary passed to #init-acronyms(dict)")
+          }else{
+            acronyms.at(acr).at(0)
+          }
+        }else{
+          panic("Definitions should be arrays of one or two strings. Definition of "+acr+ " is: "+type(defs))
+        }
+      }else{
+        panic(acr+" is not a key in the acronyms dictionary.")
+      }
+      
+      })
+  }
+}
+
+#let acr(acr) = {
+  // Display an acronym in the singular form. Expands it if used for the first time.
+  
+  // Generate the key associated with this acronym
+  let state-key = "acronym-state-" + acr
+  // Create a state to keep track of the expansion of this acronym
+  state(state-key,false).display(seen => {if seen{acr}else{[#display-def(plural: false, acr) (#acr)]}})
+  state(state-key,false).update(true)
+}
+
+#let acrpl(acr) = {
+  // Display an acronym in the plural form. Expands it if used for the first time. 
+    
+  // Generate the key associated with this acronym
+  let state-key = "acronym-state-" + acr
+  // Create a state to keep track of the expansion of this acronym
+  state(state-key,false).display(seen => {if seen{acr+"s"}else{[#display-def(plural: true, acr) (#acr\s)]}})
+  state(state-key,false).update(true)
+}
+
+#let print-index(level: 1, outlined: false, sorted:"", title:"Acronyms Index") = {
+  //Print an index of all the acronyms and their definitions.
+  // Args:
+  //   level: level of the heading. Default to 1.
+  //   outlined: make the index section outlined. Default to false
+  //   sorted: define if and how to sort the acronyms: "up" for alphabetical order, "down" for reverse alphabetical order, "" for no sort (print in the order they are defined). If anything else, sort as "up". Default to ""
+  //   title: set the title of the heading. Default to "Acronyms Index". Passing an empty string will result in removing the heading.
+
+  // assert on input values to avoid cryptic error messages
+  assert(sorted in ("","up","down"), message:"Sorted must be a string either \"\", \"up\" or \"down\"")
+
+  if title != ""{
+    heading(level: level, outlined: outlined)[#title]
```

Comment:
Could also expose other heading parameters like `numbering` to be able to adjust the styling of the index's heading?

Reply:
My idea was that if you want a special heading, remove it from the return by passing `title: ""`and then make your own heading exactly as you want. Do that not work for you?

Reply:
Ah, sure, that also works :) Thank you!